### ISSUE TITLE
Automapper 8.0 is not compatible with current infrastructure

### ISSUE THREAD

**Issue by quigamdev** [riganti/infrastructure#44]
Automapper 8.0 brought breaking changes and infrastructure is not compatible with that. 
Method IMemberConfigurationExpression.ResolveUsing was removed. 

http://docs.automapper.org/en/stable/8.0-Upgrade-Guide.html

**Comment by LadislavBohm** [comment]
@tomasherceg Sorry to ping you like this but can someone have a look at PR #52 fixing this? It's been over a year and referencing this old version of AutoMapper can make it hard for apps to transition later. Thanks

**Comment by tomasherceg** [comment]
@LadislavBohm Sorry about that, I completely missed out the PR. I'll try to review it soon.

**Comment by tomasherceg** [comment]
@LadislavBohm Just want to let you know that we've been looking at the issue with @quigamdev but found some problems with AutoMapper we cannot solve right away. We are not giving up, but it's still in progress.
 
Basically, the problem is that the new version of AutoMapper removed the static methods `Mapper.Map` and so on. It's not the problem because we had abstracted this away using `IEntityDTOMapper`. However, there was the `ProjectTo` extension method for queryables that didn't require any parameters, but now it needs `IMapperConfiguration`. 
Unfortunately, most users use `ProjectTo` in query objects, and updating to new AutoMapper means that they'll have to add a constructor argument to all of their queries in order to be able to call `ProjectTo` - we tried it on a real app and found out that we need to modify more than 100 classes.

So we are looking for a way how to solve this - we are considering to write a Roslyn analyzer that would be able to fix this.
